crates/wasmtime/src/engine.rs

```diff
@@ -519,7 +520,7 @@ impl Engine {
     pub fn precompile_module(&self, bytes: &[u8]) -> Result<Vec<u8>> {
         #[cfg(feature = "wat")]
         let bytes = wat::parse_bytes(&bytes)?;
-        let (v, _) = crate::compile::build_artifacts::<Vec<u8>>(self, &bytes)?;
+        let (v, _) = crate::compile::build_artifacts::<Vec<u8>>(self, &bytes, None)?;
```

Comment:
Could this change to using the `ModuleBuilder` API and have this implementation move to within the `ModuleBuilder`?